### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by farukcankaya**
- Decrease the minimum required Python version
- Simplify the example
- Update configurations and remove unnecessary configs, tools, etc.
- Add `black` and `isort` code formatting tools

### CODE REVIEW

**Review by alumarcu — CHANGES_REQUESTED** [review]
A few other things left from my earlier feedback:
- Regarding the example directory:
    - In `example` - poetry and pyproject don't need to exist. The end-developers might choose to use requirements.txt instead
    or something else entirely
    - I would suggest adding a README.md inside the Example directory, explaining the structure.
- `pyproject.toml` - Try to explain the library in layman's terms and a bit more concisely.
- Keywords, in lowercase could be: `dependencies`, `architecture`, `jdepend`, `deptrac` (the last two are names of equivalent java and php tools)

File: .flake8 [hunk 1]
```diff
@@ -1,7 +1,5 @@
 [flake8]
-max-line-length = 99
-max-doc-length = 99
+max-line-length = 199
+max-doc-length = 199
```
[the remark this answers is not in the document]
Comment (farukcankaya) [reply .flake8]: We use 119 in the main codebase, I changed it to the same for this repo.

**Review by alumarcu — APPROVED** [review]
